tlvc/src/lib.rs

```diff
@@ -321,13 +350,22 @@ impl<R> ChunkHandle<R> {
         R: TlvcRead,
     {
         let end = position
-            .checked_add(u64::try_from(dest.len()).unwrap())
+            .checked_add(
+                u64::try_from(dest.len())
+                    .ok()
+                    .ok_or(TlvcReadError::Truncated)?,
```

Comment:
`ok().ok_or(TlvcReadError::Truncated)?` feels like a kind of strange way to spell:
```suggestion
                    .map_err(|_| TlvcReadError::Truncated)?,
```

Perhaps the `ok()` was added before the error type change, or something?

Reply:
Mmm, no good reason, yeah. I was trying to be consistent with the Option-ok branches but as you say, it's definitely more than a little strange.

Thank you! <3